Sources/TuistHasher/ResourcesContentHasher.swift

```diff
@@ -3,43 +3,78 @@ import TuistCore
 import XcodeGraph
 
 public protocol ResourcesContentHashing {
-    func hash(resources: ResourceFileElements) throws -> String
+    func hash(identifier: String, resources: ResourceFileElements) throws -> MerkleNode
 }
 
 /// `ResourcesContentHasher`
 /// is responsible for computing a unique hash that identifies a list of resources
 public final class ResourcesContentHasher: ResourcesContentHashing {
     private let contentHasher: ContentHashing
     private let privacyManifestContentHasher: PrivacyManifestContentHasher
+    private let platformConditionContentHasher: PlatformConditionContentHashing
 
     // MARK: - Init
 
     public convenience init(contentHasher: ContentHashing) {
         self.init(
             contentHasher: contentHasher,
-            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher)
+            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher),
+            platformConditionContentHasher: PlatformConditionContentHasher(contentHasher: contentHasher)
         )
     }
 
     public init(
         contentHasher: ContentHashing,
-        privacyManifestContentHasher: PrivacyManifestContentHasher
+        privacyManifestContentHasher: PrivacyManifestContentHasher,
+        platformConditionContentHasher: PlatformConditionContentHashing
     ) {
         self.contentHasher = contentHasher
         self.privacyManifestContentHasher = privacyManifestContentHasher
+        self.platformConditionContentHasher = platformConditionContentHasher
     }
 
     // MARK: - ResourcesContentHashing
 
-    public func hash(resources: ResourceFileElements) throws -> String {
-        var hashes = try resources.resources
-            .sorted(by: { $0.path < $1.path })
-            .map { try contentHasher.hash(path: $0.path) }
+    public func hash(identifier: String, resources: ResourceFileElements) throws -> MerkleNode {
+        var children: [MerkleNode] = []
+
+        children
+            .append(
+                contentsOf: try resources.resources.sorted(by: { $0.path < $1.path })
+                    .map { try hashResourceFileElement(element: $0) }
+            )
```

Comment:
```suggestion
        var children: [MerkleNode] = try resources.resources.sorted(by: { $0.path < $1.path })
                .map { try hashResourceFileElement(element: $0) }
```

Why not assign the resources merkle nodes directly?